sycl/include/sycl/ext/oneapi/experimental/graph.hpp

```diff
@@ -31,6 +31,44 @@ namespace oneapi {
 namespace experimental {
 
 namespace detail {
+// List of sycl features and extensions which are not supported by graphs. Used
+// for throwing errors when these features are used with graphs.
+enum class UnsupportedGraphFeatures {
+  sycl_reductions,
+  sycl_specialization_constants,
+  sycl_kernel_bundle,
+  sycl_ext_oneapi_kernel_properties,
+  sycl_ext_oneapi_enqueue_barrier,
+  sycl_ext_oneapi_memcpy2d,
+  sycl_ext_oneapi_device_global,
+  sycl_ext_oneapi_bindless_images
```

Comment:
I would prefer if we give these explicit integer values. That way, if they ever cross library boundaries, we don't end up with mixups if someone adds a new value in the middle of the enum. Unless they explicitly change the other values, of course, but then hopefully we catch them in the act.